test/bench.cpp

```diff
@@ -1,90 +1,392 @@
+                    if (temp_it != temp_inputs.end()) {
+                        matrix.back().emplace_back(temp[j]);
+                        temp.erase(temp.begin() + j);
+                        j--;
+                    }
+                }
+            }
+            layer_num++;
+        }
+    temp.clear();
+
+    for (const auto &iter : stats.element) {
+        if (iter.type == "output") {
+            temp.emplace_back(iter);
+        }
+    }
+    matrix.emplace_back(temp);
+    temp.clear();
+}
+
+int maxStr(std::vector<std::vector<parametrs>> &matrix) {
+    size_t max = 0;
+    for (const auto &iter : matrix) {
+        if (max < iter.size()) max = iter.size();
+    }
+    return max;
+}
+
+void calculate_fp_coordinates( std::vector<parametrs>::iterator &it_str,
+                               std::vector<parametrs>::iterator &el_pred_str, std::vector<std::vector<parametrs>> &matrix) {
+    int fcoef_scale = 4;
+    int scoef_scale = 3;
+    it_str->fp_x = el_pred_str->sp_x + fcoef_scale * SCREEN_WIDTH / (2*scoef_scale*matrix.size());
+    it_str->fp_y = el_pred_str->sp_y + SCREEN_WIDTH / (4*matrix.size());
+
+}
+
+void calculate_sp_coordinates( std::vector<parametrs>::iterator &it_str,
+                               std::vector<std::vector<parametrs>>::iterator &it_matrix, std::vector<std::vector<parametrs>> &matrix) {
+    int indent = 10;
+    it_str->sp_x = indent + (SCREEN_WIDTH/matrix.size()) * (it_matrix - matrix.begin());
+    it_str->sp_y = indent + (SCREEN_HEIGHT / maxStr(matrix)) * (it_str - it_matrix->begin());
+
+}
+
+void setCoordinates(std::vector<std::vector<parametrs>> &matrix) {
+    for (auto it_matrix = matrix.begin(); it_matrix != matrix.end(); ++it_matrix) {
+        for (auto it_str = it_matrix->begin(); it_str != it_matrix->end(); ++it_str) {
+            it_str->coeff_x = it_matrix - matrix.begin();
+            it_str->coeff_y = it_str - it_matrix->begin();
+
+            if (it_str->type == "input"){
+                calculate_sp_coordinates(it_str, it_matrix, matrix);
+            }
+
+            else{
+                for (auto el_pred_str = (it_matrix-1)->begin(); el_pred_str != (it_matrix-1)->end(); ++el_pred_str){
+
+                    auto result = std::find(it_str->inputs.begin(), it_str->inputs.end(), el_pred_str->output);
+                    if (result != it_str->inputs.end()) {
+                        calculate_fp_coordinates(it_str, el_pred_str, matrix);
+                        calculate_sp_coordinates(it_str, it_matrix, matrix);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void matrix_to_vector(std::vector<parametrs> &vector, std::vector<std::vector<parametrs>> &matrix) {
+    for (const auto &i : matrix){
+        for (const auto &j : i){
+            vector.emplace_back(j);
+        }
+    }
+}
+
+void debug(bench_statistics &stats, std::vector<parametrs> &vector_of_elements ){
```

Comment:
It is okay until debug output is required on something else than `stderr`: it is better to pass an output as a parameter.